mirar/processors/sources/namer.py

```diff
@@ -91,28 +99,40 @@ def get_next_name(self, detection_time: Time, last_name: str = None) -> str:
         :return: new name
         """
         cand_year = detection_time.datetime.year % 1000
+
+        if last_name is not None:
+            last_year = self.extract_last_year(last_name)
+            if last_year != cand_year:
+                last_name = None
+
         if last_name is None:
-            res = run_select(
-                query=select(getattr(self.db_table.sql_model, self.db_name_field))
-                .order_by(text(f"{self.db_order_field} desc"))
-                .limit(1),
-                sql_table=self.db_table.sql_model,
+
+            col = self.db_table.sql_model.__table__.c[self.db_name_field]
+
+            # Select most recent name of same year
+            sel = (
+                select(col).where(col.contains(cand_year)).order_by(col.desc()).limit(1)
```

Comment:
Does this work where col is a string and cand_year is an int?

Reply:
cand_year is always an int in this function

Reply:
(and col is not a string it is an sqlalchemy column object)

Reply:
Yeah, I just wanted to confirm that it works because col is a column containing strings of the form 'WNTR23aaaaa' and cand_year is an int (e.g. 23).